src/Cmfd.cpp

```diff
@@ -1333,217 +1056,104 @@ void Cmfd::setNumFSRs(int num_fsrs) {
-              _surface_currents[(y*_num_x+x)*ncg*8 + 5*ncg + e];
-          _surface_currents[(y*_num_x+x+1)*ncg*8 + 1*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 5*ncg + e];
-        }
-      }
-      
-      /* Split the RIGHT TOP CORNER */
-      
-      /* If cell is not on right or top geometry edge
-       * give to right surface and top surface of mesh cell to the right */
-      if (x < _num_x - 1 && y < _num_y - 1) {
-        for (int e = 0; e < ncg; e++) {
-          log_printf(DEBUG, "cell: %d, group: %d, RIGHT TOP current: %f", 
-              y*_num_x+x,e, _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e]);
-          _surface_currents[(y*_num_x+x)*ncg*8 + 2*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-          _surface_currents[(y*_num_x+x)*ncg*8 + 3*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-          _surface_currents[(y*_num_x+x+1)*ncg*8 + 3*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-          _surface_currents[((y+1)*_num_x+x)*ncg*8 + 2*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-        }
-      }
-      /* If cell is on right geometry edge
-       * give to right surface and top surface */
-      else if (x == _num_x - 1 && y != _num_y - 1) {
-        for (int e = 0; e < ncg; e++) {
-          log_printf(DEBUG, "cell: %d, group: %d, RIGHT TOP current: %f", 
-              y*_num_x+x,e, _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e]);
-          _surface_currents[(y*_num_x+x)*ncg*8 + 2*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-          _surface_currents[(y*_num_x+x)*ncg*8 + 3*ncg + e] += 
-              _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-          _surface_currents[((y+1)*_num_x+x)*ncg*8 + 2*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-        }
-      }
-      /* If cell is on top geometry edge
-       * give to right surface and top surface */
-      else if (x != _num_x - 1 && y == _num_y - 1) {
-        for (int e = 0; e < ncg; e++) {
-          log_printf(DEBUG, "cell: %d, group: %d, RIGHT TOP current: %f", 
-              y*_num_x+x,e, _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e]);
-          _surface_currents[(y*_num_x+x)*ncg*8 + 2*ncg + e] += 
-              _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-          _surface_currents[(y*_num_x+x)*ncg*8 + 3*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
-          _surface_currents[(y*_num_x+x+1)*ncg*8 + 3*ncg + e] += 
-              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
+  #pragma omp parallel for private(current)
+  for (int y = 0; y < _num_y; y++) {
+    for (int x = 0; x < _num_x; x++) {
+      int cell = y*_num_x + x;
+      for (int s = 4; s < NUM_SURFACES; s++) {
```

Comment:
That sounds good.
On Aug 2, 2015 8:13 AM, "sam" <notifications@github.com> wrote:

> In src/Cmfd.cpp
> <https://github.com/mit-crpg/OpenMOC/pull/182#discussion_r36044121>:
>
> > -       * give to right surface and top surface */
> > -      else if (x != _num_x - 1 && y == _num_y - 1) {
> > -        for (int e = 0; e < ncg; e++) {
> > -          log_printf(DEBUG, "cell: %d, group: %d, RIGHT TOP current: %f",
> > -              y*_num_x+x,e, _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e]);
> > -          _surface_currents[(y*_num_x+x)*ncg*8 + 2*ncg + e] +=
> > -              _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
> > -          _surface_currents[(y*_num_x+x)*ncg*8 + 3*ncg + e] +=
> > -              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
> > -          _surface_currents[(y*_num_x+x+1)*ncg*8 + 3*ncg + e] +=
> > -              0.5 * _surface_currents[(y*_num_x+x)*ncg*8 + 6*ncg + e];
> > +  #pragma omp parallel for private(current)
> > +  for (int y = 0; y < _num_y; y++) {
> > +    for (int x = 0; x < _num_x; x++) {
> > +      int cell = y*_num_x + x;
> > +      for (int s = 4; s < NUM_SURFACES; s++) {
>
> As it stands now, the enum values that represent the corners need to start
> at 4 and go up from there in order to correctly index into the
> _surface_currents vector. One option could be to add a _corner_currents
> vector and have each segment store a cmfd surface for each direction and a
> cmfd corner for each direction. This would separate the surface currents
> and corner currents and allow this loop to be more intuitive. What do you
> think about that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/OpenMOC/pull/182/files#r36044121>.
>